### ISSUE TITLE
Extension not shown as running in Codespace, although it's installed

### ISSUE THREAD

**Issue by bamurtaugh** [microsoft/vscode#119908]
**Version:** 

Codespaces web

Version: 1.55.0-insider
Commit: 6ebe2a14f3722bc2582ecd8108d0a71dfdd46036
Date: 2021-03-25T16:22:19.936Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36

**Steps to Reproduce:**

1. This has happened twice now when attempting to use the GHPRI Nightly extension in Codespaces. I specifically check I have the most recent Nightly version of the extension installed, but when I actually go to use it, it's pulling some outdated version, or something with the extension glitches
2. I check **Developer: Show Running Extensions**, and GitHub Pull Requests Nightly is not listed
3. I refresh my Codespace, and now GHPRI is behaving as expected, and **Developer: Show Running Extensions** lists it

I also exported logs from the Codespace and can share them in case they help: 
[logs (5).zip](https://github.com/microsoft/vscode/files/6207444/logs.5.zip)

**Screenshots:**

The Codespace initially had an incorrect/outdated version of the extension:

![image](https://user-images.githubusercontent.com/25310137/112535283-5873d200-8d69-11eb-9db5-ed2758d0b4a8.png)

GHPRI Nightly wasn't listed in my running extensions:

![image](https://user-images.githubusercontent.com/25310137/112535343-6c1f3880-8d69-11eb-9e62-334a79f60b8d.png)

After reloading my Codespace, it picks up the correct version of GHPRI Nightly:

![image](https://user-images.githubusercontent.com/25310137/112535406-80633580-8d69-11eb-9228-13248c0090d6.png)

cc @RMacfarlane

**Comment by sandy081** [comment]
May I know if you have both GHPR stable and nightly versions installed? Do you have GHPR stable disabled? I am wondering why any GHPR extension is not listed in running extensions but you can still see the UI from that extension.

**Comment by bamurtaugh** [comment]
I had the Stable version of GHPR disabled:

![image](https://user-images.githubusercontent.com/25310137/112655655-dab5d200-8e0d-11eb-865c-24a53397f09f.png)

I was also unsure how I could have any UI from that extension if neither the Stable nor the Insiders GHPR is shown in running extensions.

**Comment by sandy081** [comment]
@alexdima Any insights why the GHPR nightly extension is not shown in running extensions list even if it is running? If we can see that we can know which version of it is running.

**Comment by alexdima** [comment]
The Running Extensions List only shows extensions which have resolved their activation promise, i.e. which are considered "activated". It doesn't show extensions that are currently activating. So in this case, based on the screenshots, I believe the GHPR got activated, and is still activating (e.g. its `activate` promise did not resolve) and that is why it does not appear in the running extensions list.

@bamurtaugh I have looked through the logs, but those logs only appear to capture the "service side" of things. Can you perhaps get the remote extension host log? (`File > Open > /home/codespace/.vscode-remote/data/logs/<timestamp>/exthostX/exthost.log`). This will help to see if the GHPR got activated and my hunch above is correct.

@sandy081 I think this is a very interesting case. I speculate that the GHPR is installed by the codespace itself as a builtin extension, and that occurs before the settings sync has a chance to run and fetch the enablement state. So I think the GHPR is activated before settings sync tells us that it should be disabled.

**Comment by bamurtaugh** [comment]
@alexdima I found several levels of extension host logs:

![image](https://user-images.githubusercontent.com/25310137/112917932-8e41ef00-90b8-11eb-8eec-59707ca66228.png)

I clicked through all 4 options here, and `exthost1` and `exthost2` had their own `exthost.log` files. Here are their respective contents:

[exthost1.txt](https://github.com/microsoft/vscode/files/6225440/exthost1.txt)
[exthost2.txt](https://github.com/microsoft/vscode/files/6225441/exthost2.txt)

I can see in exthost1: 

> Activating extension GitHub.vscode-pull-request-github-insiders failed due to an error:
> [2021-03-25 19:26:15.553] [exthost] [error] Error: GitHub Pull Requests and Issues Nightly cannot be used while GitHub Pull Requests and Issues is also installed. Please ensure that only one version of the extension is installed.

This is interesting because when I just re-reviewed the video recording of this session, I had the Nightly extension installed, and the Stable was disabled (I didn't have Stable already enabled or try to install/enable both extensions simultaneously myself).

Please let me know if this helps or if other log files would be interesting.

**Comment by sandy081** [comment]
> I think this is a very interesting case. I speculate that the GHPR is installed by the codespace itself as a builtin extension, and that occurs before the settings sync has a chance to run and fetch the enablement state. So I think the GHPR is activated before settings sync tells us that it should be disabled.

Yeah very good hunch. I only block for initializing global state and settings. Enablement state comes with extensions which is pulled only after some time. I will split extensions initialization into two parts

1. Syncing existing extensions state (enablement & state)
2. Syncing newly installed extensions

**Comment by sandy081** [comment]
It was already implemented - https://github.com/microsoft/vscode/commit/0772410c7d596e901de00b885e92bbd21bdb3d11
